src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java

```diff
@@ -52,19 +51,18 @@ public void processQuestionnaireLinked(
 
     Case caze = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
 
-    if (checkRequiredFieldsForIndividualHI(
-        questionnaireId, caze.getCaseType(), uac.getIndividualCaseId())) {
-      caze =
-          caseService.prepareIndividualResponseCaseFromParentCase(
-              caze, UUID.fromString(uac.getIndividualCaseId()));
+    throwExceptionIfArgumentsInvalid(caze, questionnaireId, uac.getIndividualCaseId());
```

Comment:
I'm not a big fan of `throwException` type stuff... it smacks of test code. I would call it `checkArgumentsValid`